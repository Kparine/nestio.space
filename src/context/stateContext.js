import React, { createContext, useReducer, useMemo } from "react";
import {
	SET_DATA_ACTION,
	SET_NOTIFICATION_ACTION,
	SET_AVERAGE_ACTION,
	INCREMENT_WARNING_ACTION,
	RESET_WARNING_TIME_ACTION,
	INCREMENT_SAFE_ACTION,
	RESET_SAFE_TIME_ACTION,
	SET_PREV_LOW_ACTION,
} from "../constants/action-constants";

const initialState = {
	notification: [],
	satData: [],
	avg: NaN,
	warningTime: 0,
	safeTime: 0,
	prevLow: false,
};
const StateContext = createContext(initialState);
export { StateContext };

const reducer = (state, action) => {
	switch (action.type) {
		case SET_DATA_ACTION: {
			return {
				...state,
				satData: [...state.satData, action.payload],
			};
		}
		case SET_NOTIFICATION_ACTION: {
			return {
				...state,
				notification: [...state.notification, action.payload],
			};
		}
		case SET_AVERAGE_ACTION:
			return {
				...state,
				avg: action.payload,
			};
		case INCREMENT_WARNING_ACTION:
			return {
				...state,
				warningTime: state.warningTime + action.payload,
			};
		case RESET_WARNING_TIME_ACTION:
			return {
				...state,
				warningTime: 0,
			};
		case INCREMENT_SAFE_ACTION:
			return {
				...state,
				safeTime: state.safeTime + action.payload,
			};
		case RESET_SAFE_TIME_ACTION:
			return {
				...state,
				safeTime: 0,
			};
		case SET_PREV_LOW_ACTION:
			return {
				...state,
				prevLow: action.payload,
			};
		default:
			throw new Error();
	}
};

export const StateContextProvider = ({ children }) => {
	const [state, dispatch] = useReducer(reducer, initialState);

	const contextValue = useMemo(
		() => ({
			state,
			dispatch,
		}),
		[state, dispatch]
	);

	return (
		<StateContext.Provider value={contextValue}>
			{children}
		</StateContext.Provider>
	);
};
