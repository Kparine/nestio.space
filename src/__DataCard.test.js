import React from "react";
import { mount } from "enzyme";
import { StateContext, StateContextProvider } from "./context/stateContext";
import DataCard from "./components/Data-Card/data-card";
import DataAvg from "./components/Data-Card/avg-data";
import DataCurr from "./components/Data-Card/curr-data";

describe("<DataCard />", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(
			<StateContextProvider>
				<DataCard />
			</StateContextProvider>
		);
	});
	console.log("StateContext ******------>>>>>>", StateContext);

	it("should exist and load context", () => {
		expect(wrapper).not.toBeNull();
		expect(wrapper).toHaveLength(1);
	});
	it("match snapshot", () => {
		expect(wrapper).toMatchSnapshot();
	});
	it("should exist and render DataAvg Component", () => {
		expect(wrapper.find(DataAvg)).not.toBeNull();
		expect(wrapper.find(DataAvg)).toHaveLength(1);
	});
	it("should exist and render DataCurr Component", () => {
		expect(wrapper.find(DataCurr)).not.toBeNull();
		expect(wrapper.find(DataCurr)).toHaveLength(1);
	});
});
