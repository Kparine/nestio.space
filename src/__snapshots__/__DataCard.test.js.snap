// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataCard /> match snapshot 1`] = `ReactWrapper {}`;
